### ISSUE TITLE
Multiple Contexts shutdown leak or not clear

### ISSUE THREAD

**Issue by zlnimda** [ocornut/imgui#1558]
Hello, I am using imgui and my app has multiple windows. 
_The version used is one year old but the weird problem is still present in last version._
I used as recommended since [there](https://github.com/ocornut/imgui/commit/8b428e8c74c013bd84cf4284e69192c0e222a4c9) the use of multiple context, one by windows.

I hooked our memory management system on Alloc & Free callbacks and it seems there was some **leaks**.
At first my code was looking like that:
```c++
auto defaultContext = ImGui::GetCurrentContext(); 
[...]
ImGui::CreateContext(...); // multiple times 
[...]
ImGui::DestroyContext(...); // multiple times
[...]
ImGui::SetCurrentContext(defaultContext);
ImGui::Shutdown();
```

I found that each allocation by context was not freed, so I changed my code in this :
```c++
auto defaultContext = ImGui::GetCurrentContext(); 
[...]
wContext = ImGui::CreateContext(...); // multiple times
[...]
ImGui::SetCurrentContext(wContext);
ImGui::Shutdown();
ImGui::DestroyContext(...);
[...]
ImGui::SetCurrentContext(defaultContext);
ImGui::Shutdown();
```

But it crashed because It seems that I use a single Font which is shared between context. 
```c++
// imgui.cpp line:716
static ImFontAtlas      GImDefaultFontAtlas;
// imgui.cpp line:796
ImGuiIO::ImGuiIO()
{
    Fonts = &GImDefaultFontAtlas;
}
```

So now everything is _working_ (no leak anymore) with this change:
```diff
ImGui::SetCurrentContext(wContext);
+ ImGui::GetIO().Fonts = null;
ImGui::Shutdown();
```

But this looks so unexpected and not documented in comments. I'm not sure this is the way it should be uninitialized, so I prefer to refer my code here to know what it is the real purpose of `Shutdown()` and `DestroyContext()`. The destructor `~ImGuiContext` doesn't exist but is explicitly called in `DestroyContext()`. 

Please let me know what you think about this case.

**Comment by ocornut** [comment]
Thank you for the detailed post.

There are several issues related to this (#591, #1007, #586, #992). 

I don't have the exact answer right now but those issues have been lingering and I will try to tackle some of them soon.


Also note that I am working on a new system to use a single imgui context for multiple windows (see #1542). If you are not looking for multi-threading or rendering at different frequency, using that new system will be much better :)

**Comment by ocornut** [comment]
@zlnimda With the changes included in #1565 (now merged) all the issues here should be solved :)
